From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Danny Lin <danny@kdrag0n.dev>
Date: Mon, 11 Oct 2021 19:24:58 -0700
Subject: [PATCH 10/24] SystemUI: Follow light/dark theme in quick settings

Android 12's dual-tone style where the quick settings panel is always
dark makes the light theme look like a second-class citizen. Pure black
also looks out-of-place next to QS tiles and the notification panel
because dynamic themes don't affect it at all.

Revert to the ~Beta 1 style where quick settings used the same theme as
the notification shade.

- colorAccentPrimary has been replaced with colorAccent for contrast in
  light mode, because colorAccentPrimary is system_accent1_100 (dark
  accent color)
- Footer chips have been converted to surfaces (similar to QS tiles and
  notifications), which makes more sense with the new style
- The QS background is now the same as the notification shade background
  in both light and dark modes

Demo screenshots (with dual-tone commit):
https://twitter.com/kdrag0n/status/1445922541218922496

[kdrag0n: ported to 12L]
[neobuddy89: adapted for 14QPR1]

Saalim:
- Use colorAccents in shade values to match systme theme design change

Change-Id: I3d45b72f0f119e100505409d178ab8f698993881
Signed-off-by: Pranav Vashi <neobuddy89@gmail.com>
Signed-off-by: Saalim Quadri <danascape@gmail.com>
[a17: light QS colours combined with the /e/ day/night e_qs aliases; the night
 Theme.SystemUI.QuickSettings is rebuilt from the Android 17 day style (only
 colorError differs); ScrimState uses a17 mBouncerSurfaceColor]

(cherry picked from commit 1b491f4f4016acafd5f72216d744945afaf1b330)
---
 packages/SystemUI/res/values-night/styles.xml | 32 ++++++++++++++++++-
 packages/SystemUI/res/values/styles.xml       | 14 ++++----
 .../statusbar/phone/ScrimController.java      |  2 +-
 .../systemui/statusbar/phone/ScrimState.java  | 16 +++++-----
 4 files changed, 47 insertions(+), 17 deletions(-)

diff --git a/packages/SystemUI/res/values-night/styles.xml b/packages/SystemUI/res/values-night/styles.xml
index 2b9ca4051fff..40f227721a0a 100644
--- a/packages/SystemUI/res/values-night/styles.xml
+++ b/packages/SystemUI/res/values-night/styles.xml
@@ -14,7 +14,8 @@
      limitations under the License.
 -->
 
-<resources xmlns:android="http://schemas.android.com/apk/res/android">
+<resources xmlns:android="http://schemas.android.com/apk/res/android"
+           xmlns:androidprv="http://schemas.android.com/apk/prv/res/android">
 
     <style name="Theme.SystemUI.DayNightDialog" parent="@android:style/Theme.DeviceDefault.Dialog"/>
 
@@ -24,6 +25,35 @@
         <item name="android:windowIsFloating">true</item>
     </style>
 
+    <!-- /e/OS: QS follows light/dark theme; same as the day style except colorError -->
+    <style name="Theme.SystemUI.QuickSettings" parent="@*android:style/Theme.DeviceDefault">
+        <item name="isQsTheme">true</item>
+        <item name="lightIconTheme">@style/QSIconTheme</item>
+        <item name="darkIconTheme">@style/QSIconTheme</item>
+        <item name="android:colorError">@*android:color/error_color_material_dark</item>
+        <item name="android:windowIsFloating">true</item>
+        <item name="android:homeAsUpIndicator">@drawable/ic_arrow_back</item>
+
+        <item name="surfaceBright">@color/materialColorSurfaceBright</item>
+        <item name="android:colorBackground">?attr/surfaceBright</item>
+        <item name="scHigh">@color/materialColorSurfaceContainerHigh</item>
+        <item name="primary">@color/materialColorPrimary</item>
+        <item name="tertiary">@color/materialColorTertiary</item>
+        <item name="onSurface">@color/materialColorOnSurface</item>
+        <item name="onSurfaceVariant">@color/materialColorOnSurfaceVariant</item>
+        <item name="outline">@color/materialColorOutline</item>
+
+        <item name="shadeActive">?android:attr/colorAccent</item>
+        <item name="onShadeActive">?android:attr/textColorPrimaryInverse</item>
+        <item name="onShadeActiveVariant">?android:attr/textColorSecondaryInverse</item>
+        <item name="shadeInactive">@*android:color/e_qs_tile_inactive</item>
+        <item name="onShadeInactive">?android:attr/textColorPrimary</item>
+        <item name="onShadeInactiveVariant">?android:attr/textColorSecondary</item>
+        <item name="shadeDisabled">@*android:color/e_qs_tile_inactive</item>
+        <item name="underSurface">@*android:color/e_qs_background</item>
+        <item name="android:itemTextAppearance">@style/Control.MenuItem</item>
+    </style>
+
     <!-- Screenshots -->
     <style name="LongScreenshotActivity" parent="@android:style/Theme.DeviceDefault.DayNight">
         <item name="android:windowNoTitle">true</item>
diff --git a/packages/SystemUI/res/values/styles.xml b/packages/SystemUI/res/values/styles.xml
index 436ae93afb17..5bbebc918d81 100644
--- a/packages/SystemUI/res/values/styles.xml
+++ b/packages/SystemUI/res/values/styles.xml
@@ -520,7 +520,7 @@
         <item name="isQsTheme">true</item>
         <item name="lightIconTheme">@style/QSIconTheme</item>
         <item name="darkIconTheme">@style/QSIconTheme</item>
-        <item name="android:colorError">@*android:color/error_color_material_dark</item>
+        <item name="android:colorError">@*android:color/error_color_material_light</item>
         <item name="android:windowIsFloating">true</item>
         <item name="android:homeAsUpIndicator">@drawable/ic_arrow_back</item>
 
@@ -533,13 +533,13 @@
         <item name="onSurfaceVariant">@color/materialColorOnSurfaceVariant</item>
         <item name="outline">@color/materialColorOutline</item>
 
-        <item name="shadeActive">@color/customColorShadeActive</item>
-        <item name="onShadeActive">@color/customColorOnShadeActive</item>
-        <item name="onShadeActiveVariant">@color/customColorOnShadeActiveVariant</item>
+        <item name="shadeActive">?android:attr/colorAccent</item>
+        <item name="onShadeActive">?android:attr/textColorPrimaryInverse</item>
+        <item name="onShadeActiveVariant">?android:attr/textColorSecondaryInverse</item>
         <item name="shadeInactive">@*android:color/e_qs_tile_inactive</item>
-        <item name="onShadeInactive">@color/customColorOnShadeInactive</item>
-        <item name="onShadeInactiveVariant">@color/customColorOnShadeInactiveVariant</item>
-        <item name="shadeDisabled">@color/customColorShadeDisabled</item>
+        <item name="onShadeInactive">?android:attr/textColorPrimary</item>
+        <item name="onShadeInactiveVariant">?android:attr/textColorSecondary</item>
+        <item name="shadeDisabled">@*android:color/e_qs_tile_inactive</item>
         <item name="underSurface">@*android:color/e_qs_background</item>
         <item name="android:itemTextAppearance">@style/Control.MenuItem</item>
     </style>
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/ScrimController.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/ScrimController.java
index 7b68d0c46994..e105a9e67909 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/ScrimController.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/ScrimController.java
@@ -1125,7 +1125,7 @@ public class ScrimController implements ViewTreeObserver.OnPreDrawListener, Dump
                 mNotificationsAlpha = behindAlpha;
                 mNotificationsTint = behindTint;
                 mBehindAlpha = 1;
-                mBehindTint = Color.BLACK;
+                mBehindTint = Color.TRANSPARENT;
             } else {
                 mBehindAlpha = behindAlpha;
                 if (mState == ScrimState.KEYGUARD && mTransitionToFullShadeProgress > 0.0f) {
diff --git a/packages/SystemUI/src/com/android/systemui/statusbar/phone/ScrimState.java b/packages/SystemUI/src/com/android/systemui/statusbar/phone/ScrimState.java
index 6c20ebc23225..62aadb632b36 100644
--- a/packages/SystemUI/src/com/android/systemui/statusbar/phone/ScrimState.java
+++ b/packages/SystemUI/src/com/android/systemui/statusbar/phone/ScrimState.java
@@ -99,7 +99,7 @@ public enum ScrimState {
                 mBehindAlpha = mClipQsScrim ? 1 : mScrimBehindAlphaKeyguard;
                 mNotifAlpha = mClipQsScrim ? mScrimBehindAlphaKeyguard : 0;
                 if (mClipQsScrim) {
-                    updateScrimColor(mScrimBehind, 1f /* alpha */, mBackgroundColor);
+                    updateScrimColor(mScrimBehind, 1f /* alpha */, Color.TRANSPARENT);
                 }
             }
         }
@@ -129,7 +129,7 @@ public enum ScrimState {
                 return;
             }
             mBehindAlpha = mClipQsScrim ? 1 : mDefaultScrimAlpha;
-            mBehindTint = mClipQsScrim ? mBackgroundColor : mBouncerSurfaceColor;
+            mBehindTint = mBouncerSurfaceColor;
             mNotifAlpha = mClipQsScrim ? mDefaultScrimAlpha : 0;
             mNotifTint = Color.TRANSPARENT;
             mFrontAlpha = 0f;
@@ -212,10 +212,10 @@ public enum ScrimState {
                 mBehindAlpha = mClipQsScrim ? 1 : mDefaultScrimAlpha;
                 mNotifAlpha = 1f;
                 mFrontAlpha = 0f;
-                mBehindTint = mClipQsScrim ? Color.TRANSPARENT : mBackgroundColor;
+                mBehindTint = Color.TRANSPARENT;
 
                 if (mClipQsScrim) {
-                    updateScrimColor(mScrimBehind, 1f /* alpha */, mBackgroundColor);
+                    updateScrimColor(mScrimBehind, 1f /* alpha */, Color.TRANSPARENT);
                 }
             }
         }
@@ -327,17 +327,17 @@ public enum ScrimState {
             mAnimateChange = !mOccludeAnimationPlaying && !fromAod;
 
             mFrontTint = Color.TRANSPARENT;
-            mBehindTint = mBackgroundColor;
+            mBehindTint = Color.TRANSPARENT;
             mBlankScreen = false;
 
             if (mDisplayRequiresBlanking && previousState == ScrimState.AOD) {
                 // Set all scrims black, before they fade transparent.
                 updateScrimColor(mScrimInFront, 1f /* alpha */, mBackgroundColor /* tint */);
-                updateScrimColor(mScrimBehind, 1f /* alpha */, mBackgroundColor /* tint */);
+                updateScrimColor(mScrimBehind, 1f /* alpha */, Color.TRANSPARENT /* tint */);
 
                 // Scrims should still be black at the end of the transition.
                 mFrontTint = mBackgroundColor;
-                mBehindTint = mBackgroundColor;
+                mBehindTint = Color.TRANSPARENT;
                 mBlankScreen = true;
             } else if (Flags.notificationShadeBlur()) {
                 mBehindTint = mShadePanelColor;
@@ -349,7 +349,7 @@ public enum ScrimState {
             }
 
             if (mClipQsScrim) {
-                updateScrimColor(mScrimBehind, 1f /* alpha */, mBackgroundColor);
+                updateScrimColor(mScrimBehind, 1f /* alpha */, Color.TRANSPARENT);
             }
         }
 
